### DISCUSSION TITLE
Unable to restore backup: specified backup location doesn't exist/isn't a file

### DISCUSSION THREAD

**Discussion by Prokypter** [mrrfv/open-android-backup#71] [Q&A] [answered]
Hi! I've been having some trouble restoring to my phone using the script. Backup worked flawlessly and is accessible (thank you very much) but I've been unable to select a file to restore at that part of the script as it doesn't seem to be happy about the .7z backup archive. Advice on how to restore/get the files to be recognised would be much appreciated. I'm on an up-to-date Arch installation with all dependencies installed. 
![image](https://github.com/mrrfv/open-android-backup/assets/92405070/637dbafd-bbb9-483f-a876-beec874de7ad)

**Comment by Prokypter** [comment]
No syntax errors have occurred to my knowledge. I've tried the drag-and-drop method, renaming manually, and a few others, but no luck each time.

**Comment by mrrfv** [comment]
Could you please make sure that you are providing the full location to the backup archive, there are no spaces in it and the script has access to the file?

If that doesn't work, you could try directly setting the path to the backup using the `archive_path` environment variable. For example: `archive_path="/home/me/backup.7z" ./backup.sh`

As a last resort, try editing the script (`./functions/restore_func.sh`) so it doesn't check if the directories are valid.

**Reply by Prokypter** [reply, marked as answer]
Thank you for the quick reply! I've narrowed down the problem to being with the environment variables I choose: the script works when given the archive_path variable, but ONLY when I use your example script given on GitHub (password filled in on both). If I try with only archive_path and archive_password, I get the error. No other combination of variables thus far has given a working script, including your suggestion just now. Could I have missed an important variable or step? Again, syntax appears correct, I'm not missing quotes and the full path is provided each time. 

Just to clarify, I've now got it working- thank you! Looking to debug this for future use though.

**Comment by dev590t** [comment]
I have a similar issue than you.

The backup path is `/tmp`. So I enter `/tmp` as restore path. Then I get this error. The correct path is the 7z archive contained in `/tmp`: `/tmp/open-android-backup-12-07-2023-19-59-21.7z`, and that run.
